devnulli/EvlWatcher
# Testing the V2 Beta

Hi.

Testing the version 2.0
Please write your feedback directly into that pull request.

# Testing Release Candidate (from 2020-11-14 to 2020-12-01)

**until 2020-12-01** After that, it will be merged to `master` and a release will be made

# Testing Beta 2 - Frontend things (from 2020-11-07 2020-11-14)

**until 2020-11-14** After this week, it will be merged, and `v2_release_candidate` will be brought up in a new pull request.

- Revamped the console
  - Live output of what the service is doing
  - Rule tester (using Eventlog XML)
  - See and control the banlist, whitelist, blacklist
  - configure the logging  
  - reload button to reload configuration from config file

- Fixed a security leak concerning the named pipe between Console and Service

**how to test beta 2:**

- play around with the console
- should not work when not elevated (admin)
- settings you take (global or tasks) should be saved it the config.xml
- play around with the rule tester
- try to create your own rule
- hackers-> try to issue commands via the named pipe `net.pipe://localhost/EvlWatcher` from non-privileged users

# Testing Beta 1 - Backend things (from 2020-11-02 to 2020-11-07)

**until 2020-11-07**, please try out the [Beta 1](https://github.com/devnulli/EvlWatcher/blob/v2-beta1/Versions/v2/EvlWatcher-v2.0%20beta%201-setup.exe), a while. Focus on Basic functionality (Installer, Service Running, Bans working, RAM usage ok). After this week, it will be merged, and `v2_beta2` will be brought up in a new pull request.

- EvlWatcher 2.0 is now running on .NetFramework 4.7.2, which is the default framework for Windows servers (2019, ver1809, ver1803) -> see https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/versions-and-dependencies
- EvlWatcher 2.0 is now using a better software architecture pattern, making it easier to change and adapt in the future
- EvlWatcher 2.0 is now using a different configuration file structure
- EvlWatcher 2.0 should fix the ram usage problem of the previous versions
- EvlWatcher 2.0 BETA 1 is released without a console

**How to test beta 1:**

- Install EvlWatcher
	- there should be no need to uninstall previous versions, it will be replaced. if that doesnt work, please report that in the pull request ✅ 

- you can set the loglevel in the config.xml to "Debug" for a full output. Just dont leave it that way, your eventlog will grow. ℹ️ 

- once you have the impression it runs stable, leave it running for a few days, to find out if it still uses massive amounts of RAM over the time, and of course, if it still bans.
  - memory leak was still there 🟨 (in test)

- Evlwatcher should still recognize IPs out the event log, according to config.xmlif there are any adjustments to be made to these rules, also let me know please.
  - rule 4625 didnt work ✅ 

- If you find additional event log entries where EvlWatcher does not react, please don't just report, also post a **SAMPLE**, especially with information on where the ip could be extracted. ℹ️ 

- uninstalling should completely remove evlwatcher (no service, no folder in programs(x86))
 ✅ 

**In case something crashes or doesnt work:**
EvlWatcher writes everything into an own Eventsource called Evlwatcher.
It should be easy to make a view to show these events.
In case something crashed, please post the error message as well.

![image](https://user-images.githubusercontent.com/3720480/97917135-4f4f7d80-1d54-11eb-9dba-1e27971ac3fb.png)

Thanks for your help in testing EvlWatcher.
